sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -164,28 +140,12 @@ case class DataSourceResolution(
         s"Delete from tables is not supported using the legacy / v1 Spark external catalog" +
             s" API. Identifier: $table.")
 
-    case delete: DeleteFromStatement =>
-      val relation = UnresolvedRelation(delete.tableName)
-      val aliased = delete.tableAlias.map(SubqueryAlias(_, relation)).getOrElse(relation)
-      DeleteFromTable(aliased, delete.condition)
+    case ShowNamespacesStatement(None, pattern) if defaultCatalog.isEmpty =>
```

Comment:
I found this problem while refactoring the catalog lookup logic. If no catalog is specified and default catalog is not set, we should fallback to v1 command which is `ShowDatabasesCommand` in this case.

Reply:
OK, I may have misunderstood this discussion: https://github.com/apache/spark/pull/25601#discussion_r319379649 - I had an impression that `SHOW NAMESPACES` is supported only in v2.

Reply:
I don't think this is correct. The `SHOW DATABASES` command was not changed, so using `SHOW DATABASES` is the v1 fallback. `SHOW NAMESPACES` always uses v2. I think that's fine.